src/res_core.ml

```diff
@@ -1391,12 +1412,12 @@ and parseRecordPatternField p =
 and parseRecordPatternItem p =
   match p.Parser.token with
   | DotDotDot ->
-    Parser.next p;
-    Some (true, PatField (parseRecordPatternField p))
+    Parser.expect DotDotDot p;
+    Some (true, PatField (parseRecordPatternField ~label:(parseValuePath p) p))
   | Uident _ | Lident _ ->
-    Some (false, PatField (parseRecordPatternField p))
+    Some (false, PatField (parseRecordPatternField ~label:(parseValuePathNotEof p) p))
```

Comment:
Here we do know the token is not `Eof`. If we did not propagate this info, which lets us use `next` instead of `nextUnsafe` termination analysis would fire.